### ISSUE TITLE
CmdStanMCMC `summary` method - don't specify `sig_figs` unless use does

### ISSUE THREAD

**Issue by mitzimorris** [stan-dev/cmdstanpy#550]
#### Summary:
The call to the `bin/stansummary` function always has `sig_figs` specified - default is 2, but this
can result in losing precision if intermediate CSV file created by `bin/stansummary` uses scientific notation,
at which numbers like `4441` become `4400`.

#### Description:
reports on Discourse here:  https://discourse.mc-stan.org/t/cmdstanpy-error-in-fit-summary/26939

#### Current Version:
1.0.1

**Comment by WardBrian** [comment]
So do we simply not want to set a default value here? Does not passing `--sig-figs=2` lead to a different result than passing it, since 2 is the default value?

**Comment by mitzimorris** [comment]
it's unclear what to do here - the other option is trying to fix things on the CmdStan side - the way that the "save to CSV" option works is not good - if the output is in scientific notation, you only get 2 digit's worth of precision, not 2 decimal places.   it would be preferable to fix this on the CmdStan side.

**Comment by WardBrian** [comment]
Should we open a new issue there instead?

**Comment by mitzimorris** [comment]
yes - it would be good to figure out what's going on with the CSV.

**Comment by WardBrian** [comment]
It seems that the CSV output doesn't respect the default sig-figs that the docs cite.

without specifying `-s`:
```csv
name,Mean,MCSE,StdDev,5%,50%,95%,N_Eff,N_Eff/s,R_hat
"lp__",-7.29137,0.0194958,0.751762,-8.82028,-6.99659,-6.74977,1486.9,19062.8,0.999907
```

with `-s 2`
```csv
name,Mean,MCSE,StdDev,5%,50%,95%,N_Eff,N_Eff/s,R_hat
"lp__",-7.3,0.019,0.75,-8.8,-7,-6.7,1.5e+03,1.9e+04,1
```

I should note that the values displayed in stdout do seem to have a default value of 2 significant figures.

This is a cmdstan bug, I think.
